query_execution/WorkOrdersContainer.hpp

```diff
@@ -343,6 +346,18 @@ class WorkOrdersContainer {
            getNumRebuildWorkOrders(operator_index);
   }
 
+  /**
+   * @brief Check if the given operator has ever a normal work order.
+   *
+   * @param index The index of the given operator in the DAG.
+   *
+   * @return True if the operator has a normal work order, false otherwise.
+   **/
+  bool hasEverNormalWorkOrders(const std::size_t operator_index) const {
```

Comment:
Suggest to rename this function to ``hasEverProducedNormalWorkOrder`` to make more sense.